secure_message/services/party_service.py

```diff
@@ -10,18 +10,30 @@
 
 
 class PartyService:
+
+    users_cache = dict()
+    business_details_cache = dict()
+
     @staticmethod
-    def get_business_details(ru):
-        """Retrieves the business details from the party service"""
+    def get_url(api_param, code):
+        """
+        :param api_param:  is the key configuration that that represent part or the URI .
+        :param code: is the code to use in the url ( uuid or ru )
+        :return: a formatted url
+        """
+        if current_app.config[api_param] is None or current_app.config['RAS_PARTY_SERVICE'] is None:
+            raise KeyError("%s not present" % api_param)
```

Comment:
This would be more Pythonic:

```Python
try:
    current_app.config[api_param].format(current_app.config['RAS_PARTY_SERVICE'], code)
except KeyError:
	raise KeyError(f"{api_param} not present")
```